src/components/ApiPublishModal/OpenApiSpec.tsx

```diff
@@ -57,14 +79,84 @@ const OpenApiSpec: React.FC<IProps> = props => {
           },
         ]}
       >
-        <S.Textarea rows={15} placeholder="Enter OpenAPI Spec in YAML/JSON format" />
+        <S.Textarea
+          rows={15}
+          placeholder="Enter OpenAPI Spec in YAML/JSON format"
+          onChange={e => {
+            const spec = e.target.value;
+
+            if (!spec) {
+              setWarnings([]);
+            } else {
+              setWarnings(checkMockingExamples(YAML.parse(JSON.parse(JSON.stringify(spec)))));
```

Comment:
maybe you should use try/catch for parsing

Reply:
I don't think is needed here as the Form.Item is already catching the error from `YAML.parse` 

![image](https://user-images.githubusercontent.com/47887589/165453857-63682d6d-6d79-430b-b2aa-2f14b2466baa.png)